src/Agent.Worker/TaskConditionsEvaluator.cs

```diff
@@ -0,0 +1,50 @@
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker
+{
+    public static class TaskConditionsEvaluator
+    {
+        public static bool AreConditionsSatisfied(IList<TaskCondition> conditions, Variables variables, IExecutionContext executionContext)
```

Comment:
About that - I was wondering, was it a conscious decision to stay away from Theory. It seemed a good way of writing data driven tests - also gives accurate stats on how many variations of a testcase are broken